src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -606,6 +629,35 @@ private static bool TryParseStatusCode(string responseHeader, int statusCodeStar
             return true;
         }
 
+        private static void HandleRedirectLocationHeader(EasyRequest state, string locationValue)
+        {
+            Debug.Assert(state._isRedirect);
+            Debug.Assert(state._handler.AutomaticRedirection);
+
+            string location = locationValue.Trim();
+            //only for absolute redirects
+            if (location.StartsWith(UriSchemeHttp) || location.StartsWith(UriSchemeHttps))
+            {
+                Uri forwardUri = new Uri(location);
```

Comment:
If the location URI can't be parsed, this will throw an exception.  That's fine, but is there an expected exception that's supposed to be thrown in this case? If so, we could use TryCreate instead and then throw whatever exception is needed.